modules/tensor_mechanics/src/vectorpostprocessors/InteractionIntegral.C

```diff
@@ -125,6 +130,17 @@ InteractionIntegral::InteractionIntegral(const InputParameters & parameters)
   // set unused dimensions to zero
   for (std::size_t i = _ndisp; i < 3; ++i)
     _grad_disp[i] = &_grad_zero;
+
+  if (getParam<MaterialPropertyName>("eigenstrain_gradient") != "")
```

Comment:
are you using this comparison instead of `isParamValid` due to the way the action sets this parameter? Wouldn't this break the standalone use of this class? Or are you relying on the (not guaranteed) behavior that an invalid string parameter defaults to "" automatically (if that's even true).